gradle-guide-error-prone/src/test/java/com/palantir/gradle/guide/errorprone/taskexecution/GetProjectTest.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.guide.errorprone.taskexecution;
+
+import org.junit.jupiter.api.Test;
+
+public class GetProjectTest extends IllegalMethodCalledDuringTaskExecutionTest {
```

Comment:
I'm not a huge fan of abstract test classes tbf, they typically grow unwieldy and hard to change/evolve especially as more abstract descendants are added because it'd then require changing things up and down the hierarchy.

I typically prefer to have smaller scoped helper objects that do the things I want for me. Then you can compose different bits of behaviour by just including the relevant helper objects if need be.

I won't push super hard on it, but largely flagging my distaste for this style of testing

Reply:
I think subclassing is the correct model here because there are shared fields, but I see where you're coming from. If we want to add additional functionality, let's add it to helper objects rather than bloating `IllegalMethodCalledDuringTaskExecutionTest`.